app/src/main/java/com/example/composetemplate/repositories/LoginRepository.kt

```diff
@@ -0,0 +1,57 @@
+package com.example.composetemplate.repositories
+
+import com.example.composetemplate.data.models.local_models.User
+import com.example.composetemplate.data.remote.base.BaseRequest
+import com.example.composetemplate.utils.LoginCallback
+
+
+/**
+ * This class contain the functionality of all login auth types
+ */
+class LoginRepository(
```

Comment:
As I said let's talk about it and the itercator

Reply:
same concept as AuthInteractor, make explained comment